### DISCUSSION TITLE
Returning expressions from a custom helper

### DISCUSSION THREAD

**Discussion by shimil** [handlebars-lang/handlebars.js#1997] [Q&A]
Hello,
I implemented a custom localization helper, which gets a translation key and maps it to a value e.g.
`<a href="www.some-link.com">{$t my.key}</a>`

I would like my translation values to contain other mustache expressions, such as resolving a variable from the context. An example would be
`"my.key": "click here for help with {{productName}}"`

I would like for that returned value to then be treated as an expression and resolved. Is there a way to achieve that?

Many thanks.

**Comment by jaylinski** [comment]
You could do something like this:
https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7Ba%7D%7D%20%7B%7Bt%20%5C%22foo%5C%22%7D%7D%5Cn%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20a%3A%20%5C%22value1%5C%22%2C%5Cn%20%20b%3A%20%5C%22value2%5C%22%5Cn%7D%5Cn%22%2C%22output%22%3A%22value1%20bar%20value2%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('t'%2C%20function%20(key%2C%20context)%20%7B%5Cn%20%20const%20translations%20%3D%20%7B%20foo%3A%20%5C%22bar%20%7B%7Bb%7D%7D%5C%22%20%7D%3B%5Cn%20%20%5Cn%20%20if%20(key%20in%20translations)%20%7B%5Cn%20%20%20%20const%20template%20%3D%20Handlebars.compile(translations%5Bkey%5D)%3B%5Cn%20%20%20%20return%20template(context%5B'data'%5D%5B'root'%5D)%3B%5Cn%20%20%7D%5Cn%20%20%20%5Cn%20%20return%20key%3B%5Cn%7D)%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.8%22%7D

```js
Handlebars.registerHelper('t', function (key, context) {
  const translations = { foo: "bar {{b}}" };
  
  if (key in translations) {
    const template = Handlebars.compile(translations[key]);
    return template(context['data']['root']);
  }
   
  return key;
})
```

Maybe there is a better way...

**Reply by shimil** [reply]
Thanks @jaylinski.
The problem with this solution is that I want to take advantage of the different partials and other helpers that I have registered with Handlebars elsewhere.

**Reply by satyajitnayk** [reply]
@shimil you can do something like
```ts
 Handlebars.registerHelper(Helpers); 
 // where Helpers is List of custom helpers & partials you have defined somewhere else 
 // imported here/in the same file you doing this!!
```
before doing `Handlebar.compile()`
